nmstate-varlink/nmstate_varlink.py

```diff
@@ -0,0 +1,236 @@
+            },
+        )
+
+
+class NmstateError(varlink.VarlinkError):
+    def __init__(self, message):
+        varlink.VarlinkError.__init__(
+            self,
+            {
+                "error": "io.nmstate.NmstateError",
+                "parameters": {"reason": str(message)},
+            },
+        )
+
+
+class NmstateLibnmError(varlink.VarlinkError):
+    def __init__(self, message):
+        varlink.VarlinkError.__init__(
+            self,
+            {
+                "error": "io.nmstate.NmstateLibnmError",
+                "parameters": {"reason": str(message)},
+            },
+        )
+
+
+class NmstateVerificationError(varlink.VarlinkError):
+    def __init__(self, message):
+        varlink.VarlinkError.__init__(
+            self,
+            {
+                "error": "io.nmstate.NmstateVerificationError",
+                "parameters": {"reason": str(message)},
+            },
+        )
+
+
+class NmstateNotImplementedError(varlink.VarlinkError):
+    def __init__(self, message):
+        varlink.VarlinkError.__init__(
+            self,
+            {
+                "error": "io.nmstate.NmstateNotImplementedError",
+                "parameters": {"reason": str(message)},
+            },
+        )
+
+
+class NmstateInternalError(varlink.VarlinkError):
+    def __init__(self, message):
+        varlink.VarlinkError.__init__(
+            self,
+            {
+                "error": "io.nmstate.NmstateInternalError",
+                "parameters": {"reason": str(message)},
+            },
+        )
+
+
+class NmstateDependencyError(varlink.VarlinkError):
+    def __init__(self, message):
+        varlink.VarlinkError.__init__(
+            self,
+            {
+                "error": "io.nmstate.NmstateDependencyError",
+                "parameters": {"reason": str(message)},
+            },
+        )
+
+
+class ServiceRequestHandler(varlink.RequestHandler):
+    service = service
+
+
+@service.interface("io.nmstate")
+class NmstateService:
+    def Show(self, include_status_data):
+        """
+        Reports the state data on the system
+        """
+        if include_status_data is None:
+            include_status_data = False
+        try:
+            lib_show = libnmstate.show(include_status_data=include_status_data)
+            state = {
+                "state": {
+                    DNS.KEY: lib_show[DNS.KEY],
+                    Interface.KEY: lib_show[Interface.KEY],
+                    RouteRule.KEY: lib_show[RouteRule.KEY],
+                    Route.KEY: lib_show[Route.KEY],
+                }
+            }
+            return state
+        except libnmstate.error.NmstateValueError as exception:
+            raise NmstateValueError(str(exception))
+
+    def Apply(self, state, verify_change, commit, rollback_timeout):
+        """
+        Apply desired state declared in json format
+        """
+        if verify_change is None:
```

Comment:
Please do not repeat the defaults from libnmstate.apply() here. Maybe you can do something like

```
apply_kwargs = {}
if verify_change is not None:
    apply_kwargs["verify_change"] = verify_change
...
libnmstate.apply(state, **kwargs)
```

Not sure, if there is a better way. Ideally, Apply would be `def Apply(self, state, **kwargs)` and then you could just pass `kwargs`.